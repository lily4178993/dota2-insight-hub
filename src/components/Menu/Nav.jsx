import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import './nav.css';

const Nav = () => {
  const [isNavOpen, setIsNavOpen] = useState(false);
  return (
    <nav className="nav">
      <div className="nav-container">
        <div
          className="navbar"
          style={{ background: isNavOpen ? '' : 'var(--color-blue-4)' }}
        >
          <button
            type="button"
            title={isNavOpen ? 'close' : 'menu'}
            className="menu-toggle"
            onClick={() => setIsNavOpen(!isNavOpen)}
          >
            <div className={isNavOpen ? 'hamBox hamBoxOpen' : 'hamBox'}>
              <span className={isNavOpen ? 'lineTop spin' : 'lineTop'}> </span>
              <span className={isNavOpen ? 'lineBottom spin' : 'lineBottom'}>
                {' '}
              </span>
            </div>
          </button>
          <div className="nav-settings">
            <i className="bx bx-microphone" />
            <i className="bx bx-cog" />
          </div>
        </div>
        <div
          className="nav-overlay"
          style={{
            top: isNavOpen ? '0' : '-100%',
            transitionDelay: isNavOpen ? '0s' : '0s',
          }}
        >
          <ul className="nav-links">
            <li className="nav-item">
              <Link
                to="/"
                rel="noopener noreferrer"
                aria-label="Home"
                onClick={() => setIsNavOpen(!isNavOpen)}
                style={{
                  top: isNavOpen ? '0' : '120px',
                  transitionDelay: isNavOpen ? '0.8s' : '0s',
                }}
              >
                Home
              </Link>
              <div className="nav-item-wrapper"> </div>
            </li>
            <li className="nav-item">
              <Link
                to="/"
                rel="noopener noreferrer"
                aria-label="Heroes"
                onClick={() => setIsNavOpen(!isNavOpen)}
                style={{
                  top: isNavOpen ? '0' : '120px',
                  transitionDelay: isNavOpen ? '0.9s' : '0s',
                }}
              >
                Heroes
              </Link>
              <div className="nav-item-wrapper"> </div>
            </li>
            <li className="nav-item">
              <Link
                to="/"
                rel="noopener noreferrer"
                aria-label="Home"
                onClick={() => setIsNavOpen(!isNavOpen)}
                style={{
                  top: isNavOpen ? '0' : '120px',
                  transitionDelay: isNavOpen ? '1s' : '0s',
                }}
              >
                Items
              </Link>
              <div className="nav-item-wrapper"> </div>
            </li>
            <li className="nav-item">
              <Link
                to="/"
                rel="noopener noreferrer"
                aria-label="Pro Matches"
                onClick={() => setIsNavOpen(!isNavOpen)}
                style={{
                  top: isNavOpen ? '0' : '120px',
                  transitionDelay: isNavOpen ? '1.1s' : '0s',
                }}
              >
                Pro Matches
              </Link>
              <div className="nav-item-wrapper"> </div>
            </li>
            <li className="nav-item">
              <Link
                to="/"
                rel="noopener noreferrer"
                aria-label="Pro Players"
                onClick={() => setIsNavOpen(!isNavOpen)}
                style={{
                  top: isNavOpen ? '0' : '120px',
                  transitionDelay: isNavOpen ? '1.2s' : '0s',
                }}
              >
                Pro Players
              </Link>
              <div className="nav-item-wrapper"> </div>
            </li>
          </ul>
          <div className="nav-footer">
            <div className="dev-infos">
              <Link
                to="/"
                rel="noopener noreferrer"
                aria-label="About Me"
                onClick={() => setIsNavOpen(!isNavOpen)}
                style={{
                  top: isNavOpen ? '0' : '-20px',
                  opacity: isNavOpen ? '1' : '0',
                  transitionDelay: isNavOpen ? '1.2s' : '0s',
                }}
              >
                About Me
              </Link>
              <Link
                to="/"
                rel="noopener noreferrer"
                aria-label="References"
                onClick={() => setIsNavOpen(!isNavOpen)}
                style={{
                  top: isNavOpen ? '0' : '-20px',
                  opacity: isNavOpen ? '1' : '0',
                  transitionDelay: isNavOpen ? '1.3s' : '0s',
                }}
              >
                References
              </Link>
            </div>
            <div className="nav-social-media">
              <ul>
                <li>
                  <a
                    href="#Github"
                    target="_blank"
                    onClick={() => setIsNavOpen(!isNavOpen)}
                    style={{
                      top: isNavOpen ? '0' : '-20px',
                      opacity: isNavOpen ? '1' : '0',
                      transitionDelay: isNavOpen ? '1.4s' : '0s',
                    }}
                  >
                    Github
                  </a>
                </li>
                <li>
                  <a
                    href="#Twitter"
                    target="_blank"
                    onClick={() => setIsNavOpen(!isNavOpen)}
                    style={{
                      top: isNavOpen ? '0' : '-20px',
                      opacity: isNavOpen ? '1' : '0',
                      transitionDelay: isNavOpen ? '1.5s' : '0s',
                    }}
                  >
                    Twitter
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default Nav;
