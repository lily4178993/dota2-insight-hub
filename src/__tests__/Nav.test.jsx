import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Nav from '../components/Menu/Nav';

test('Nav component matches snapshot', () => {
  const component = renderer.create(
    <Router>
      <Nav />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
