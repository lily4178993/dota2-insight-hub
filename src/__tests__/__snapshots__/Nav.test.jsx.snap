// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<nav
  className="nav"
>
  <div
    className="nav-container"
  >
    <div
      className="navbar"
      style={
        Object {
          "background": "var(--color-blue-4)",
        }
      }
    >
      <button
        className="menu-toggle"
        onClick={[Function]}
        title="menu"
        type="button"
      >
        <div
          className="hamBox"
        >
          <span
            className="lineTop"
          >
             
          </span>
          <span
            className="lineBottom"
          >
             
          </span>
        </div>
      </button>
      <div
        className="nav-settings"
      >
        <i
          className="bx bx-microphone"
        />
        <i
          className="bx bx-cog"
        />
      </div>
    </div>
    <div
      className="nav-overlay"
      style={
        Object {
          "top": "-100%",
          "transitionDelay": "0s",
        }
      }
    >
      <ul
        className="nav-links"
      >
        <li
          className="nav-item"
        >
          <a
            aria-label="Home"
            href="/"
            onClick={[Function]}
            rel="noopener noreferrer"
            style={
              Object {
                "top": "120px",
                "transitionDelay": "0s",
              }
            }
          >
            Home
          </a>
          <div
            className="nav-item-wrapper"
          >
             
          </div>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-label="Heroes"
            href="/details/heroes"
            onClick={[Function]}
            rel="noopener noreferrer"
            style={
              Object {
                "top": "120px",
                "transitionDelay": "0s",
              }
            }
          >
            Heroes
          </a>
          <div
            className="nav-item-wrapper"
          >
             
          </div>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-label="Home"
            href="/details/items"
            onClick={[Function]}
            rel="noopener noreferrer"
            style={
              Object {
                "top": "120px",
                "transitionDelay": "0s",
              }
            }
          >
            Items
          </a>
          <div
            className="nav-item-wrapper"
          >
             
          </div>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-label="Pro Matches"
            href="/details/proMatches"
            onClick={[Function]}
            rel="noopener noreferrer"
            style={
              Object {
                "top": "120px",
                "transitionDelay": "0s",
              }
            }
          >
            Pro Matches
          </a>
          <div
            className="nav-item-wrapper"
          >
             
          </div>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-label="Pro Players"
            href="/details/proPlayers"
            onClick={[Function]}
            rel="noopener noreferrer"
            style={
              Object {
                "top": "120px",
                "transitionDelay": "0s",
              }
            }
          >
            Pro Players
          </a>
          <div
            className="nav-item-wrapper"
          >
             
          </div>
        </li>
      </ul>
      <div
        className="nav-footer"
      >
        <div
          className="dev-infos"
        >
          <a
            aria-label="About Me"
            href="/aboutme"
            onClick={[Function]}
            rel="noopener noreferrer"
            style={
              Object {
                "opacity": "0",
                "top": "-20px",
                "transitionDelay": "0s",
              }
            }
          >
            About Me
          </a>
          <a
            aria-label="References"
            href="/references"
            onClick={[Function]}
            rel="noopener noreferrer"
            style={
              Object {
                "opacity": "0",
                "top": "-20px",
                "transitionDelay": "0s",
              }
            }
          >
            References
          </a>
        </div>
        <div
          className="nav-social-media"
        >
          <ul>
            <li>
              <a
                aria-label="GitHub page"
                href="https://github.com/lily4178993/dota2-insight-hub.git"
                onClick={[Function]}
                rel="noreferrer"
                style={
                  Object {
                    "opacity": "0",
                    "top": "-20px",
                    "transitionDelay": "0s",
                  }
                }
                target="_blank"
              >
                Github
              </a>
            </li>
            <li>
              <a
                aria-label="LinkedIn page"
                href="#LinkedIn"
                onClick={[Function]}
                rel="noreferrer"
                style={
                  Object {
                    "opacity": "0",
                    "top": "-20px",
                    "transitionDelay": "0s",
                  }
                }
              >
                LinkedIn
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
